Subject: [PATCH] aea
---
Index: src/base/tools/aapt2/link/ReferenceLinker.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/base/tools/aapt2/link/ReferenceLinker.cpp b/src/base/tools/aapt2/link/ReferenceLinker.cpp
--- a/src/base/tools/aapt2/link/ReferenceLinker.cpp	(revision 584338ea98273df82df5bc33ba137a0470cc82d6)
+++ b/src/base/tools/aapt2/link/ReferenceLinker.cpp	(date 1770832000006)
@@ -299,10 +299,10 @@
     return nullptr;
   }
 
-  if (!IsSymbolVisible(*symbol, reference, callsite)) {
-    if (out_error) *out_error = "is private";
-    return nullptr;
-  }
+  //if (!IsSymbolVisible(*symbol, reference, callsite)) {
+  //  if (out_error) *out_error = "is private";
+  //  return nullptr;
+  //}
   return symbol;
 }
 
Index: src/base/tools/aapt2/ResourceUtils.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/base/tools/aapt2/ResourceUtils.cpp b/src/base/tools/aapt2/ResourceUtils.cpp
--- a/src/base/tools/aapt2/ResourceUtils.cpp	(revision 584338ea98273df82df5bc33ba137a0470cc82d6)
+++ b/src/base/tools/aapt2/ResourceUtils.cpp	(date 1770832000072)
@@ -222,7 +222,7 @@
     }
 
     if (!type.empty() && type != "attr") {
-      return false;
+      // Apktool: Don't die out if private resource.
     }
 
     if (entry.empty()) {
Index: src/base/tools/aapt2/Resources.proto
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/base/tools/aapt2/Resources.proto b/src/base/tools/aapt2/Resources.proto
--- a/src/base/tools/aapt2/Resources.proto	(revision 584338ea98273df82df5bc33ba137a0470cc82d6)
+++ b/src/base/tools/aapt2/Resources.proto	(date 1770832000034)
@@ -16,7 +16,7 @@
 
 syntax = "proto3";
 
-import "frameworks/base/tools/aapt2/Configuration.proto";
+import "Configuration.proto";
 
 package aapt.pb;
 
@@ -636,4 +636,4 @@
 message UntranslatableSection {
   uint64 start_index = 1;
   uint64 end_index = 2;
-}
\ No newline at end of file
+}
Index: src/base/tools/aapt2/ResourcesInternal.proto
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/base/tools/aapt2/ResourcesInternal.proto b/src/base/tools/aapt2/ResourcesInternal.proto
--- a/src/base/tools/aapt2/ResourcesInternal.proto	(revision 584338ea98273df82df5bc33ba137a0470cc82d6)
+++ b/src/base/tools/aapt2/ResourcesInternal.proto	(date 1770832000047)
@@ -16,8 +16,8 @@
 
 syntax = "proto3";
 
-import "frameworks/base/tools/aapt2/Configuration.proto";
-import "frameworks/base/tools/aapt2/Resources.proto";
+import "Configuration.proto";
+import "Resources.proto";
 
 package aapt.pb.internal;
 
Index: src/base/tools/aapt2/link/PrivateAttributeMover.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/base/tools/aapt2/link/PrivateAttributeMover.cpp b/src/base/tools/aapt2/link/PrivateAttributeMover.cpp
--- a/src/base/tools/aapt2/link/PrivateAttributeMover.cpp	(revision 584338ea98273df82df5bc33ba137a0470cc82d6)
+++ b/src/base/tools/aapt2/link/PrivateAttributeMover.cpp	(date 1770831999995)
@@ -81,7 +81,6 @@
     }
 
     ResourceTableType* priv_attr_type = package->FindOrCreateType(ResourceType::kAttrPrivate);
-    CHECK(priv_attr_type->entries.empty());
     priv_attr_type->entries = std::move(private_attr_entries);
   }
   return true;
Index: src/base/tools/aapt2/ResourceTable.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/base/tools/aapt2/ResourceTable.cpp b/src/base/tools/aapt2/ResourceTable.cpp
--- a/src/base/tools/aapt2/ResourceTable.cpp	(revision 584338ea98273df82df5bc33ba137a0470cc82d6)
+++ b/src/base/tools/aapt2/ResourceTable.cpp	(date 1770832000061)
@@ -460,9 +460,9 @@
   const bool validate = validation_ == Validation::kEnabled;
   const Source source = res.value ? res.value->GetSource() : Source{};
   if (validate && !res.allow_mangled && !IsValidResourceEntryName(res.name.entry)) {
-    diag->Error(DiagMessage(source)
+    diag->Warn(DiagMessage(source)
                 << "resource '" << res.name << "' has invalid entry name '" << res.name.entry);
-    return false;
+    return true;
   }
 
   if (res.id.has_value() && !res.id->first.is_valid()) {
Index: src/base/tools/aapt2/java/JavaClassGenerator.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/base/tools/aapt2/java/JavaClassGenerator.cpp b/src/base/tools/aapt2/java/JavaClassGenerator.cpp
--- a/src/base/tools/aapt2/java/JavaClassGenerator.cpp	(revision 584338ea98273df82df5bc33ba137a0470cc82d6)
+++ b/src/base/tools/aapt2/java/JavaClassGenerator.cpp	(date 1770831999984)
@@ -58,6 +58,8 @@
     "true",       "false",        "null"};
 
 static bool IsValidSymbol(const StringPiece& symbol) {
+  // Apktool: Everything is a valid symbol
+  return true;
   return sJavaIdentifiers.find(symbol) == sJavaIdentifiers.end();
 }
 
Index: src/base/tools/aapt2/cmd/Link.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/base/tools/aapt2/cmd/Link.cpp b/src/base/tools/aapt2/cmd/Link.cpp
--- a/src/base/tools/aapt2/cmd/Link.cpp	(revision 584338ea98273df82df5bc33ba137a0470cc82d6)
+++ b/src/base/tools/aapt2/cmd/Link.cpp	(date 1770831999967)
@@ -844,7 +844,20 @@
         return false;
       }
 
-      if (zip_collection->FindFile(kProtoResourceTablePath) != nullptr) {
+
+      if (zip_collection->FindFile(kApkResourceTablePath) != nullptr) {
+        std::unique_ptr<LoadedApk> app_apk = LoadedApk::LoadBinaryApkFromFileCollection(
+          Source(path), std::move(zip_collection), context_->GetDiagnostics());
+        if (app_apk == nullptr) {
+          context_->GetDiagnostics()->Error(
+    DiagMessage(path) << "This file doesn't contain a package resource table");
+        }
+        ResourceTable* table = app_apk->GetResourceTable();
+
+        context_->GetExternalSymbols()->AppendSource(
+            util::make_unique<ResourceTableSymbolSource>(table));
+        static_library_includes_.push_back(std::move(app_apk));
+      } else if (zip_collection->FindFile(kProtoResourceTablePath) != nullptr) {
         // Load this as a static library include.
         std::unique_ptr<LoadedApk> static_apk = LoadedApk::LoadProtoApkFromFileCollection(
             Source(path), std::move(zip_collection), context_->GetDiagnostics());
@@ -2326,9 +2339,9 @@
     if (package_id_int > std::numeric_limits<uint8_t>::max()
         || package_id_int == kFrameworkPackageId
         || (!options_.allow_reserved_package_id && package_id_int < kAppPackageId)) {
-      context.GetDiagnostics()->Error(
+      context.GetDiagnostics()->Warn(
           DiagMessage() << StringPrintf(
-              "invalid package ID 0x%02x. Must be in the range 0x7f-0xff.", package_id_int));
+              "invalid package ID 0x%02x. Must be in the range 0x7f-0xff. Ignoring...", package_id_int));
       return 1;
     }
     context.SetPackageId(static_cast<uint8_t>(package_id_int));
Index: src/base/tools/aapt2/util/Util.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/base/tools/aapt2/util/Util.cpp b/src/base/tools/aapt2/util/Util.cpp
--- a/src/base/tools/aapt2/util/Util.cpp	(revision 584338ea98273df82df5bc33ba137a0470cc82d6)
+++ b/src/base/tools/aapt2/util/Util.cpp	(date 1770832000019)
@@ -23,7 +23,6 @@
 
 #include "android-base/stringprintf.h"
 #include "androidfw/StringPiece.h"
-#include "build/version.h"
 
 #include "text/Unicode.h"
 #include "text/Utf8Iterator.h"
@@ -231,10 +230,7 @@
   // Update minor version whenever a feature or flag is added.
   static const char* const sMinorVersion = "19";
 
-  // The build id of aapt2 binary.
-  static const std::string sBuildId = android::build::GetBuildNumber();
-
-  return android::base::StringPrintf("%s.%s-%s", sMajorVersion, sMinorVersion, sBuildId.c_str());
+  return android::base::StringPrintf("%s.%s", sMajorVersion, sMinorVersion);
 }
 
 static size_t ConsumeDigits(const char* start, const char* end) {
