Subject: [PATCH] aea
---
Index: src/base/libs/androidfw/include/androidfw/StringPiece.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/base/libs/androidfw/include/androidfw/StringPiece.h b/src/base/libs/androidfw/include/androidfw/StringPiece.h
--- a/src/base/libs/androidfw/include/androidfw/StringPiece.h	(revision 584338ea98273df82df5bc33ba137a0470cc82d6)
+++ b/src/base/libs/androidfw/include/androidfw/StringPiece.h	(date 1770832003745)
@@ -75,6 +75,11 @@
   bool operator>(const BasicStringPiece<TChar>& rhs) const;
   bool operator==(const BasicStringPiece<TChar>& rhs) const;
   bool operator!=(const BasicStringPiece<TChar>& rhs) const;
+  // for std::basic_string
+  bool operator<(const std::basic_string<TChar>& rhs) const;
+  bool operator>(const std::basic_string<TChar>& rhs) const;
+  bool operator==(const std::basic_string<TChar>& rhs) const;
+  bool operator!=(const std::basic_string<TChar>& rhs) const;
 
   const_iterator begin() const;
   const_iterator end() const;
@@ -256,6 +261,26 @@
   return compare(rhs) != 0;
 }
 
+template <typename TChar>
+inline bool BasicStringPiece<TChar>::operator<(const std::basic_string<TChar>& rhs) const {
+    return compare(BasicStringPiece(rhs)) < 0;
+}
+
+template <typename TChar>
+inline bool BasicStringPiece<TChar>::operator>(const std::basic_string<TChar>& rhs) const {
+    return compare(BasicStringPiece(rhs)) > 0;
+}
+
+template <typename TChar>
+inline bool BasicStringPiece<TChar>::operator==(const std::basic_string<TChar>& rhs) const {
+    return compare(BasicStringPiece(rhs)) == 0;
+}
+
+template <typename TChar>
+inline bool BasicStringPiece<TChar>::operator!=(const std::basic_string<TChar>& rhs) const{
+    return compare(BasicStringPiece(rhs)) != 0;
+}
+
 template <typename TChar>
 inline typename BasicStringPiece<TChar>::const_iterator BasicStringPiece<TChar>::begin() const {
   return data_;
Index: src/base/libs/androidfw/ResourceTypes.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/base/libs/androidfw/ResourceTypes.cpp b/src/base/libs/androidfw/ResourceTypes.cpp
--- a/src/base/libs/androidfw/ResourceTypes.cpp	(revision 584338ea98273df82df5bc33ba137a0470cc82d6)
+++ b/src/base/libs/androidfw/ResourceTypes.cpp	(date 1770832003774)
@@ -43,7 +43,7 @@
 #include <utils/String16.h>
 #include <utils/String8.h>
 
-#ifdef __ANDROID__
+#if 0
 #include <binder/TextOutput.h>
 
 #endif
